specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+              args.ClientCertificateRequestResponseState = CoreWebView2ClientCertificateRequestResponseState.ContinueWithoutCertificate;
+            }
+          }
+        }, null);
+      };
+      _isClientCertificateRequested = true;
+    }
+  }
+  catch (NotImplementedException exception)
+  {
+    MessageBox.Show(this, "Client Certificate Requested Failed: " + exception.Message, "Client Certificate Requested");
+  }
+}
+```
+# Remarks
+
+# API Notes
+
+# API Details
+See [API Details](#api-details) section below for API reference.
+
+## Win32 C++
+```cpp
+interface ICoreWebView2_3;
+interface ICoreWebView2ClientCertificate;
+interface ICoreWebView2CertificateAuthorityList;
+interface ICoreWebView2ClientCertificateIssuerChainList;
+interface ICoreWebView2ClientCertificateList;
+interface ICoreWebView2ClientCertificateRequestedEventArgs;
+interface ICoreWebView2ClientCertificateRequestedEventHandler;
+
+[v1_enum]
+/// Question to reviewers if COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE
+/// enum should be replaced with Cancel and Handled properties like below to match
+/// with other WebView2 API conventions ????
+/// * Cancel = true -> AbortRequest
+/// * Handled = true -> ContinueWithoutCertificate
+/// * Handled = true + SelectedCertificate -> ContinueWithCertificate
+/// * Neither -> ContinueWithDefaultDialog
+typedef enum COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE {
+  /// Cancels the certificate selection and aborts the request.
+  COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_ABORT_REQUEST,
+  /// Continue without certificate to respond to the server.
+  COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITHOUT_CERTIFICATE,
+  /// Continue with the selected certificate to respond to the server.
+  COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITH_CERTIFICATE,
+  /// Display the default client certificate selection dialog prompt to allow the user
+  /// to choose a certificate. The SelectedCertificate property is ignored in this case.
+  COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITH_DEFAULT_DIALOG,
+} COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE;
+
+[v1_enum] typedef enum COREWEBVIEW2_CLIENT_CERTIFICATE_KIND {
+  /// Specifies smart card certificate.
+  COREWEBVIEW2_CLIENT_CERTIFICATE_KIND_SMART_CARD,
+  /// Specifies PIN certificate.
+  COREWEBVIEW2_CLIENT_CERTIFICATE_KIND_PIN,
+  /// Specifies other certificate.
+  COREWEBVIEW2_CLIENT_CERTIFICATE_KIND_OTHER,
+} COREWEBVIEW2_CLIENT_CERTIFICATE_KIND;
+
+[uuid(92d41f20-8e92-11eb-8dcd-0242ac130003), object, pointer_default(unique)]
+interface ICoreWebView2_3 : ICoreWebView2_2 {
+  /// Add an event handler for the ClientCertificateRequested event.
+  /// The ClientCertificateRequested event is raised when the WebView2
+  /// is making a request to an HTTP server that needs a client certificate
+  /// for HTTP authentication and when there is atleast one mutually trusted certitifcate.
```

Comment:
```suggestion
  /// for HTTP authentication and when there is at least one mutually trusted certificate.
```

Reply:
We decided to raise the event if no mutually trusted certificates exists, removed the sentence as N/A anymore.